101-vm-full-disk-encrypted-rhel/azuredeploy.json

```diff
@@ -0,0 +1,164 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "aadClientID": {
+      "metadata": {
+        "description": "Client ID of AAD app which has permissions to KeyVault"
+      },
+      "type": "string"
+    },
+    "aadClientSecret": {
+      "metadata": {
+        "description": "Client Secret of AAD app which has permissions to KeyVault"
+      },
+      "type": "securestring"
+    },
+    "keyVaultName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the KeyVault to place the volume encryption key"
+      }
+    },
+    "keyVaultResourceGroup": {
+      "type": "string",
+      "metadata": {
+        "description": "Resource group of the KeyVault"
+      }
+    },
+    "vmName": {
```

Comment:
Is a "vmName" parameter really necessary? It should be fine to inline "[concat('rhelVM', uniqueString(resourceGroup().id))]" in the nested deployment resource below.